src/core/linalg/src/dense/4C_linalg_tensor.hpp

```diff
@@ -459,6 +459,22 @@ namespace Core::LinAlg
    */
   constexpr auto transpose(const Rank2TensorConcept auto& A);
 
+  /*!
+   * @brief Computes the dot product of two rank-1 tensors
+   *
+   * This function performs the inner product between two vectors a * b. Both sizes need to match.
+   *
+   * @tparam TensorLeft The type of the rank-1 tensor (vector).
+   * @tparam TensorRight The type of the rank-1 tensor (vector).
+   * @param a The rank-1 tensor (vector).
+   * @param b The rank-1 tensor (vector).
+   * @return double scalar resulting from the dot product.
```

Comment:
```suggestion
   * @return Scalar value resulting from the dot product.
```

It's not going to be a double necessarily, right?